dask/array/rechunk.py

```diff
@@ -387,6 +388,23 @@ def rechunk(
         raise NotImplementedError(f"Unknown rechunking method '{method}'")
 
 
+def _choose_rechunk_method(old_chunks, new_chunks, threshold=None):
+    if method := config.get("array.rechunk.method", None) is not None:
+        return method
+    try:
+        from distributed import default_client
+
+        default_client()
+    except (ImportError, ValueError):
+        return "tasks"
+
+    _old_to_new = old_to_new(old_chunks, new_chunks)
```

Comment:
It's linear in the number of chunks (i.e., the sum of chunks along each axis) and we already call it at least once for P2P and task-based shuffling during construction. A/B tests suggest that this is "fast enough", but I'm happy to iterate should this pop up in performance profiling.

Reply:
This is more UX since it's annoying if you have to wait for too long when you type arr.rechunk(...)

But not a big deal if linear, would be nice if we could only do this once but not necessary here

Reply:
I don't think there's a good way of doing it once without complicating the code and increasing the memory footprint, so I'd like to avoid that we actually know that it's a problem. If you'd like to gauge the impact, we can do a micro-benchmark with some chunks of your choosing.

Reply:
Yeah this is fine for now